### PR TITLE
Add integer type discriminator support to System.Text.Json polymorphism

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Adds support for integer type discriminators in System.Text.Json polymorphism and incorporates API review feedback for the polymorphism attributes.

Contributes to #63747.

cc @RamjotSingh

### CODE REVIEW

File: src/libraries/System.Text.Json/ref/System.Text.Json.cs
Comment (krwq) [inline src/libraries/System.Text.Json/ref/System.Text.Json.cs:987]: fallback is a single word so I think proper capitalisation is `Fallback`

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPolymorphicTypeConfiguration.cs
Comment (krwq) [inline src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPolymorphicTypeConfiguration.cs:92]: wasn't API review against fluid APIs?

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/PolymorphicTypeResolver.cs
Comment (krwq) [inline src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/PolymorphicTypeResolver.cs:48]: fix me 😄

File: src/libraries/System.Text.Json/ref/System.Text.Json.cs
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/ref/System.Text.Json.cs:987]: This is explicitly changing the name based on API review recommendations. We can discuss again today if "fall back" is one word or two words.

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/PolymorphicTypeResolver.cs
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/PolymorphicTypeResolver.cs:48]: HomeroSimpsonGIF

File: src/libraries/System.Text.Json/ref/System.Text.Json.cs
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/ref/System.Text.Json.cs:987]: cc @bartonjs

**Comment by krwq** [comment]
I'd prefer we've waited with this change after the API review and the pending contract changes so that we don't need to change this another few times...

**Review by krwq — APPROVED** [review]
Implementation-wise this looks good to me but see my comment above. I'd prefer we haven't changed this several times and rather do everything at once in next preview. I don't feel super strongly about it so approving so you can make a final call.

File: src/libraries/System.Text.Json/ref/System.Text.Json.cs
Comment (bartonjs) [reply src/libraries/System.Text.Json/ref/System.Text.Json.cs:987]: "[fallback](https://en.wiktionary.org/wiki/fallback#English)" is a noun.  "[fall back](https://en.wiktionary.org/wiki/fall_back#English)" is a verb.  Since it's being compared with "fail", a verb, the capital B is more grammatically correct.

**Pull request merged**
